packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -72,15 +97,37 @@ class _SelectableTextSelectionGestureDetectorBuilder extends TextSelectionGestur
     // Not required.
   }
 
+  @override
+  void onSingleLongTapStart(LongPressStartDetails details) {
+    if (!delegate.selectionEnabled) {
+      return;
+    }
+    renderEditable.selectWord(cause: SelectionChangedCause.longPress);
```

Comment:
A way we can avoid having these overrides in `selectable_text.dart` so we can use the logic already in `text_selection.dart` is to use `editableText.readOnly` to differentiate between `SelectableText` and `EditableText` at the `TextSelectionGestureDetector` level. Not sure if we want to do this since maybe a user of `TextField/EditableText` might set `readOnly` to `true` and question why their gestures are behaving differently.

Reply:
The main difference is that on iOS a long press on static text select word-by-word, and on editable text it moves the collapsed cursor to the position.

On all other platforms (Android) a long press selects word-by-word.